src/Microsoft.Net.Http.Headers/HttpHeaderParser.cs

```diff
@@ -85,7 +100,17 @@ public virtual bool TryParseValues(IList<string> values, out IList<T> parsedValu
             return false;
         }
 
-        public IList<T> ParseValues(IList<string> values)
+        public virtual IList<T> ParseValues(IList<string> values)
+        {
+            return ParseValues(values, strict: false);
+        }
+
+        public virtual IList<T> ParseStrictValues(IList<string> values)
+        {
+            return ParseValues(values, strict: true);
+        }
+
+        protected virtual IList<T> ParseValues(IList<string> values, bool strict)
```

Comment:
In theory. This is the base parser and some of the implementations may need to customize it.